### ISSUE TITLE
Failed Assert in WeaponKnife.cpp

### ISSUE THREAD

**Issue by trilean** [OpenXRay/xray-16#724]
When creating a new game or loading an existing savegame, OpenXRay crashes with the following message:

```
FATAL ERROR

[error] Expression    : successCount == elementsCount || successCount == 0
[error] Function      : Load
[error] File          : /srv/vcs/openxray/src/xrGame/WeaponKnife.cpp
[error] Line          : 145
[error] Description   : You need to provide all knife splash parameters or remove them all.
[error] Argument 0    : Missing fields are:
[error] Argument 1    : splash1_hits_count, splash1_pervictim_hcount, splash2_hits_count, splash_hit_divide_factor
```
This is due to the following assert:
https://github.com/OpenXRay/xray-16/blob/d9e37273494d4800caef41d6a127f8d600956019/src/xrGame/WeaponKnife.cpp#L145-L147
Simply removing it allows the game to run.

This happens both on `master` (i.e. the December 2020 Preview) and `dev` branches.

I'm on Linux/x86_64 (Gentoo) and using the game data from GOG.

**Comment by Xottab-DUTY** [comment]
Are you sure you use the latest game version? (1.6.02)
Also, did you renamed `Localization` to `localization`?

**Comment by trilean** [comment]
Whow, thanks for the extremely fast response ;-) Yes it's 1.6.02, GOG only gives you the latest version. I've simply unpacked the GOG installer using [innoextract](https://constexpr.org/innoextract/). The `localization` folder name is in lower case.

**Comment by SkyLoaderr** [comment]
Do you have `patches\xpatch_02.db` file in your game folder?

**Comment by trilean** [comment]
Yes, it's there:

```
8d8930dd58a4a2fb62cfc74baf7f57d4  patches/xpatch_02.db
```

Also, the game reports `1.6.02` in the main menu.

**Comment by trilean** [comment]
I've done one thing not by your instructions: I've extracted the gamedata to `/opt/stalker/cop` and changed my `~/.local/share/GSC Game World/S.T.A.L.K.E.R. - Call of Pripyat/fsgame.ltx` to contain:
```
$game_root$             = true| true|  /opt/stalker/cop
[...]
```

**Comment by Xottab-DUTY** [comment]
Did you extracted all the gamedata? If so, then just unpack xpatch_02.db again and paste it to `/opt/stalker/cop`, confirm the replacements.

I don't know what unpacker you have used, but, for example, SUE (S.T.A.L.K.E.R. Universal Extractor) unpacks .xdb archives in wrong order which results in gamedata containing files from 1.6.00 instead of 1.6.02.

**Comment by trilean** [comment]
Ah sorry I didn't unpack the game's `db` files. What I unpacked was the GOG installer (an `.exe` and two or three `.bin` files) using `innoextract`. That gives you the game directory with the gamedata, binaries etc., just like when you install it on windows.

Do you want me to extract the `db`s?

**Comment by Xottab-DUTY** [comment]
Ah, now I understand.
Changing fsgame.ltx won't work. Place vanilla fsgame.ltx (or our version of it) to `/opt/stalker/cop` and launch the game with -fsltx key like that: `xr_3da -fsltx /opt/stalker/cop/fsgame.ltx` (-fsltx key requires absolute path)

**Comment by trilean** [comment]
Alright it does work now. Is it possible that this has been caused by an incompatible change in `user.ltx`? I deleted the old one and compared the old `user.ltx` with the one that the game just created -- there are quite a few changes. The previous version is a few months old, when I last tried OpenXRay.

Anyway, thanks for the great support!

(There is another issue that I've had. But I'm restarting a new game now to see if it reappears. If it does, I'll open another ticket ;-) )
